macros/src/trans.rs

```diff
@@ -157,14 +157,23 @@ fn init(app: &App, main: &mut Vec<Tokens>, root: &mut Vec<Tokens>) {
 
     let mut tys = vec![quote!(#device::Peripherals)];
     let mut exprs = vec![quote!(#device::Peripherals::all())];
+    let mut ret = None;
     let mut mod_items = vec![];
 
-    if !app.resources.is_empty() {
+    // Write resources usable by `init`, if any
+
+
+    // Are there any resources that have an initializer? Those can be used by `init`.
+    let has_initialized_resources = app.resources.iter()
```

Comment:
I think this can be expressed as `app.resource.keys().any(|res| res.expr.is_some())`.